### PR TITLE
feat: add optimistic responses to block deletes

### PR DESCRIPTION
**Pull request by Kneesal**
# Description

### Issue

block delete is now blazingly fast with the implementation of block deletes

[Link to Basecamp Todo](https://3.basecamp.com/3105655/projects)

### CODE REVIEW

**Review by tataihono — CHANGES_REQUESTED** [review]
got some failing tests that need to be addressed.

**Review by tataihono — APPROVED** [review]

File: apps/journeys-admin/src/components/Editor/Slider/Settings/CanvasDetails/Properties/blocks/Image/Options/ImageOptions.tsx [hunk 1]
```diff
@@ -84,7 +84,7 @@ export function ImageOptions(): ReactElement {
     <Box sx={{ px: 4, pb: 4 }}>
       <ImageSource
         selectedBlock={imageBlock}
-        onChange={updateImageBlock}
+        onChange={async (input) => updateImageBlock(input)}
```
Comment (edmonday) [inline apps/journeys-admin/src/components/Editor/Slider/Settings/CanvasDetails/Properties/blocks/Image/Options/ImageOptions.tsx:87]:
```suggestion
        onChange={(input) => updateImageBlock(input)}
```

File: apps/journeys-admin/src/components/Editor/Slider/Settings/CanvasDetails/Properties/blocks/Video/Options/VideoOptions.tsx [hunk 2]
```diff
@@ -97,7 +97,10 @@ export function VideoOptions(): ReactElement {
   }
 
   return selectedBlock?.__typename === 'VideoBlock' ? (
-    <VideoBlockEditor selectedBlock={selectedBlock} onChange={handleChange} />
+    <VideoBlockEditor
+      selectedBlock={selectedBlock}
+      onChange={async (input) => handleChange(input)}
```
Comment (edmonday) [inline apps/journeys-admin/src/components/Editor/Slider/Settings/CanvasDetails/Properties/blocks/Video/Options/VideoOptions.tsx:102]:
```suggestion
      onChange={(input) => handleChange(input)}
```

File: apps/journeys-admin/src/components/Editor/Slider/Settings/Drawer/VideoBlockEditor/Settings/Poster/Library/VideoBlockEditorSettingsPosterLibrary.tsx [hunk 3]
```diff
@@ -359,8 +359,8 @@ export function VideoBlockEditorSettingsPosterLibrary({
         <ImageLibrary
           open={open}
           onClose={onClose}
-          onChange={handleChange}
-          onDelete={deleteImageBlock}
+          onChange={async (input) => handleChange(input)}
+          onDelete={async () => deleteImageBlock()}
```
Comment (edmonday) [inline apps/journeys-admin/src/components/Editor/Slider/Settings/Drawer/VideoBlockEditor/Settings/Poster/Library/VideoBlockEditorSettingsPosterLibrary.tsx:363]:
```suggestion
          onChange={(input) => handleChange(input)}
          onDelete={() => deleteImageBlock()}
```

**Review by edmonday — APPROVED** [review]

**Review by tataihono — DISMISSED** [review]

**Review by tataihono — APPROVED** [review]

**Review by tataihono — DISMISSED** [review]

**Review by tataihono — APPROVED** [review]

**Pull request merged**